### DISCUSSION TITLE
Multiple /userdata folders

### DISCUSSION THREAD

**Discussion by robflate** [garethgeorge/backrest#510] [Q&A] [answered]
What is the suggested method for adding multiple `/userdata` folders in docker?

Option 1 - Separate bind mounts

```
- /data:/userdata
- /drive:/userdata-drive
- /transfers:/userdata-transfers
```

Option 2 - Mount all folders inside `/userdata`

```
- /data:/userdata/data
- /drive:/userdata/drive
- /transfers:/userdata/transfers
```

On a similar subject, how are people backing up things in directories like `/etc`? Are you just adding individual bind mounts for each folder?

**Comment by garethgeorge** [comment]
Backrest is unopinionated here, so a lot of this is up to what works best for your setup (and how you’d like the paths to be organized inside of your backups). 

I mount a number of folders at / e.g. I’m using zags and a number of zpools e.g. /tank, /tank2, /tank3, etc.

**Reply by robflate** [reply]
Thanks.

Out of interest are you mounting /tank, /tank2, /tank3 etc as individual mounts (option 1) or inside the main /userdata mount (option 2)? Also, inside the container are you maintaining the host file structure? For instance, are you doing;

`- /home/robflate/documents/video/encoded:/userdata/home/robflate/documents/video/encoded`

or not

`- /home/robflate/documents/video/encoded:/userdata/encoded-video`

I understand it doesn't matter which way I do it but I'm always interested to see what others are doing.

**Reply by garethgeorge** [reply, marked as answer]
I’m mounting into / so no /userdata folder in my case :) 

I show that in the docs primarily to illustrate / remind that data directories need to be mounted, but expect it to be replaced with whatever mounts make sense. 

The only caveat is that if adding your own mounts, avoid overwriting standard Linux paths that the docker container might be using, especially if using the alpine image which ships with a number of utilities.

**Reply by robflate** [reply]
Edit: I sent this before you added the caveat to your previous message.

Are you saying you use;

```
- /tank1:/tank1
- /tank2:/tank2
- /etc:/etc
- /var:/var
```

Does that not interfere with whatever is in those folders INSIDE the container already? For instance, there is already the `/etc` and `/var` folders inside the container.

Sorry if there's something obvious I'm missing.

**Comment by noelhibbard** [comment]
Are we suggesting this?!?:
```yaml
volumes:
  - /:/
```

That just seems like a bad idea. I was considering something like this but even then I don't feel comfortable:
```yaml
volumes:
  - /:/userdata:ro
```

I love docker but I feel like docker isn't the best solution in the case of backrect/restic but I'm hoping I can be convinced otherwise. Hahaha

**Reply by garethgeorge** [reply]
Agreed that for anything where serious data integrity, preservation of file owners / permissions, etc is important you shouldn’t trust all the filesystem magic docker is using. 

A bare metal install is the safest way to use backrest / restic for your data. 

Re; mount recommendations, backrest is unopinionated. You just need to expose your data in the container following good docker practices. I do not recommend mounting over ‘/‘ and that won’t work.
